clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -299,6 +299,7 @@ def AllocaOp : CIR_Op<"alloca", [
   }];
 
   let arguments = (ins
+    Optional<CIR_IntType>:$arraySize,
```

Comment:
Please add some documentation explaining the new operand.

Reply:
I believe we can improve the abstraction here, this is more about dynamic stack allocation than about an array size, so I suggest `dynAllocSize`. Also add an extra method "isDynamic()" so we have an API to clearly differentiate from our more traditional allocas.